glslc/src/file_compiler.cc

```diff
@@ -77,6 +89,28 @@ bool FileCompiler::CompileShaderFile(const std::string& input_file,
                 << output_file_name_ << "'" << std::endl;
     }
   }
+
+  // Handle the error message for failing to deduce the shader kind.
+  if (result.GetCompilationStatus() ==
+      shaderc_compilation_status_invalid_stage) {
```

Comment:
This doesn't depends on any file writing code, I think you are right, I should move it ahead. 

Another thing is that, if we find error during writing output, should we return 'false' for to indicate the failure? I think we should return false.